lib/software_challenge_client/piece_shape.rb

```diff
@@ -76,6 +79,30 @@ def transform(rotation, flip)
     coordinates.rotate(rotation).flip(flip)
   end
 
+  # Gibt alle Kombinationen aus Rotation und Flipping zurück, welche zu einzigartigen 
+  # Koordinatenmengen dieser Form führen. 
+  # @return [Array<Transform>] Transform Structs mit Rotation r, Boolean f
+  def unique_transforms()
+    if not defined? @transformations then
+      existing_transforms = []
+
+      Rotation.each do |r|
+        [true, false].each do |f|
+          new_transform = Transform.new(r, f, transform(r, f))
+
+          if existing_transforms.all? { |t| t.coords != new_transform.coords } then
+            existing_transforms << new_transform
+          end
+        end
+      end
+
+      @transformations = existing_transforms
+      @transformations
+    else
+      @transformations
```

Comment:
why not remove the else branch and just return @transformations outside the if?

Reply:
True, that would simplify the code